account/manager.go

```diff
@@ -1679,42 +1839,122 @@ func (m *manager) signSpendTx(ctx context.Context, account *Account,
 			witnessScript, ourSig, auctioneerSig,
 		)
 		pIn.FinalScriptWitness, err = serializeWitness(witness)
+
+	default:
+		return nil, fmt.Errorf("invalid state, should never get here")
 	}
 	if err != nil {
 		return nil, fmt.Errorf("error serializing witness: %v",
 			err)
 	}
 
-	// We either have a single account input (renew, withdraw, close) that
-	// has a final script witness set (meaning that the PSBT lib sees it as
-	// complete, even if we need to fix the witness later on) or we have
-	// additional inputs (deposit) that we need to sign for now. We use the
-	// FinalizePsbt method for the additional inputs as they belong to the
-	// normal wallet and can be signed for without additional PSBT metadata
-	// fields.
-	var signedTx *wire.MsgTx
-	if signedPacket.IsComplete() {
-		err = psbt.MaybeFinalizeAll(signedPacket)
-		if err != nil {
-			return nil, fmt.Errorf("error finalizing PSBT: %v", err)
-		}
+	return signedPacket, nil
+}
 
-		signedTx, err = psbt.Extract(signedPacket)
+// signAccountMuSig2 creates the combined MuSig2 signature to spend a Taproot
+// account output through the collaborative key spend path. This sets up a
+// MuSig2 signing session on the local signer instance and then asks the
+// auctioneer to also send its partial signature.
+func (m *manager) signAccountMuSig2(ctx context.Context, account *Account,
+	spendTx *wire.MsgTx, accountOutputIndex, accountInputIdx int,
+	modifiers []Modifier, previousOutputs []*wire.TxOut) ([]byte, error) {
+
+	sessionInfo, err := poolscript.TaprootMuSig2SigningSession(
+		ctx, account.Expiry, account.TraderKey.PubKey, account.BatchKey,
+		account.Secret, account.AuctioneerKey, m.cfg.Signer,
+		&account.TraderKey.KeyLocator,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Don't leave incomplete sessions in memory.
+	cleanupSession := func() {
+		err = m.cfg.Signer.MuSig2Cleanup(ctx, sessionInfo.SessionID)
 		if err != nil {
-			return nil, fmt.Errorf("error extracting TX: %v", err)
+			log.Errorf("Error cleaning up MuSig2 session %x: %v",
+				sessionInfo.SessionID[:], err)
 		}
-	} else {
-		// We should be able to extract the final TX now, even if the
-		// witness isn't yet fully correct just yet.
-		_, signedTx, err = m.cfg.Wallet.FinalizePsbt(
-			ctx, signedPacket, "",
+	}
+	defer func() {
+		// Can't defer cleanupSession() directly since we might change
```

Comment:
nit: maybe we can bind a variable and do something like

```
var succeeded = false

defer func() {
    if succeeded {
        // nothing to do.
        return
    }
    // clean up session...
}()
```

Reply:
Ah yes, much cleaner that way.

Reply:
Simplified the whole MuSig2 signing quite a bit so we can re-use the same code on both client and server side, which made this easier as well.